openllm-core/src/openllm_core/config/configuration_baichuan.py

```diff
@@ -61,12 +63,19 @@ class GenerationConfig:
 
   def sanitize_parameters(self,
                           prompt: str,
+                          prompt_template: PromptTemplate | None = None,
```

Comment:
prompt_template should also be a string here. Best to have the same logic either in process_prompt or the logic should live under `openllm_core.prompt`

I think process_prompt should accept `PromptTemplate` object.